airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -228,7 +229,7 @@ private void doScanLibrary() {
     }
 
     private void scanFile(MediaFile file, MusicFolder musicFolder, Date lastScanned,
-                          Map<String, Integer> albumCount, Genres genres, boolean isPodcast) {
+                          Map<String, Integer> albumCount, Genres genres, boolean isPodcast, boolean ignoreSymLinks) {
```

Comment:
I don't like the fact that you added one more argument to the scanFile method. 
In this special case it seams useless as you can invoque the settings service from inside the scanFile method code bloc. I've worked a lot analyzing the structure of this service and my thoughts is it is messy. One of my futures projects would be to simplify it and introduce a plugin aware code organisation. Adding more and more arguments to methods generally leads to more mess in code.
Another suggestion I would make is to add some debug code to indicate that a special file/directory is skipped due to symlink.

Reply:
Eh, I just did it as a knee-jerk optimization in case settingsService.getIgnoreSymLinks() was a more expensive call than it should be. I tested it out and it's not, so no problem switching it over to your preferred syntax.

And added in the requested LOG.debug() calls.